crates/hotshot/src/traits/networking/web_server_network.rs

```diff
@@ -231,6 +233,182 @@ struct Inner<TYPES: NodeType> {
 
 impl<TYPES: NodeType> Inner<TYPES> {
     #![allow(clippy::too_many_lines)]
+
+    /// Handle version 0.1 transactions
+    async fn handle_tx_0_1(&self, tx: Vec<u8>, index: u64, tx_index: &mut u64) {
+        let broadcast_poll_queue = &self.broadcast_poll_queue_0_1;
+        if index > *tx_index + 1 {
+            debug!("missed txns from {} to {}", *tx_index + 1, index - 1);
+            *tx_index = index - 1;
+        }
+
+        *tx_index += 1;
```

Comment:
Are we assuming `index > *tx_index`? If so, is this equivalent to assigning `*tx_index = index`?

Reply:
From my understanding, I don't think it's the same thing. 

In the [original code](https://github.com/EspressoSystems/HotShot/blob/c953c1ee546ad2ad9b89f164c2946b04d48b1395/crates/hotshot/src/traits/networking/web_server_network.rs#L278-L289), fetching transactions from the server returns something like `(index: u64, txs: Vec<Vec<u8>>)` (originally the `txs` was also deserialized in this step, but now that happens later).

I believe the `index` is just for the very first `tx` in the block of `txs` you get from the server, and the following transactions are processed as consecutive. So `index` itself will be the same for all of them, but you increment `tx_index` as you process each transaction.

Reply:
Makes sense. Thanks for the code comment and the explanation here!